### PR TITLE
Task 4 react api

### PR DESCRIPTION
**Pull request by SBavia**
1. Task: https://github.com/rolling-scopes-school/tasks/tree/master/react/modules/module04
2. Screenshot: 
Screenshot
3. Done 09.04.2023 / deadline 10.04.2023
4. Score: 15 / 15
All done.

### CODE REVIEW

File: task_1_react_components/src/components/cardModalItem/CardModal.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        </button>
+        {errorMessage}
+        {spinner}
+        {cardToShow && <View {...cardToShow} />}
+      </div>
+    </div>
+  );
+};
+
+const View = (props: TCardModal) => {
```
Comment (bycolour2) [inline task_1_react_components/src/components/cardModalItem/CardModal.tsx:49]: Внеси в отдельный компонент

**Review by bycolour2 — COMMENTED** [review]
Хорошо!

File: task_1_react_components/src/components/cardList/CardList.tsx [hunk 2]
```diff
@@ -1,35 +1,29 @@
...
-  }, [propsCards, cards]);
+const CardList = <T extends { id?: number | string }>(props: CardListProps<T>) => {
+  const { cards, cardComponent: Card } = props;
 
   const renderItems = (arr: T[]) => {
     const items = arr.map((card) => {
       const { id, ...itemProps } = card;
-      return <Card key={id} {...itemProps} />;
+      if (props.selectCard) {
+        return <Card key={id} {...itemProps} id={id} selectCard={props.selectCard} />;
```
Comment (ParfenenkovEdit) [inline task_1_react_components/src/components/cardList/CardList.tsx:18]: <Card key={id} {...itemProps} id={id} selectCard={props.selectCard} />

File: task_1_react_components/src/hooks/http.hook.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import { useState, useCallback } from 'react';
+
+export const useHttp = () => {
```
Comment (ParfenenkovEdit) [inline task_1_react_components/src/hooks/http.hook.tsx:3]: export const useHttp = <T extends Record<string, unkonwn>> () => {..., data: T | T[] } { url }: IConfig

File: task_1_react_components/src/utils/validationFuncs.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+export const validateName = (name: string) => {
```
Comment (ParfenenkovEdit) [inline task_1_react_components/src/utils/validationFuncs.ts:1]: : {
 isValid: boolean,
 errorMessage: string
}

File: task_1_react_components/src/components/cardModalItem/CardModal.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import { TCardModal } from '../../types';
+
+import './cardModal.css';
+
+type CardProps = {
+  onClose: () => void;
+  cardId: string;
+};
+
+const CardModal = ({ onClose, cardId }: CardProps) => {
```
Comment (ParfenenkovEdit) [inline task_1_react_components/src/components/cardModalItem/CardModal.tsx:16]: useModal = (ModalComponent, config) => { onClose, onOpen, onToggle }

**Pull request closed without merging**